core/templates/pages/exploration-player-page/services/extract-image-filenames-from-model.service.ts

```diff
@@ -160,8 +168,12 @@ export class ExtractImageFilenamesFromModelService {
 
     _allHtmlInTheState.push(this._getHintsHtml(state));
 
-    if (state.interaction.solution !== null) {
-      _allHtmlInTheState.push(this._getSolutionHtml(state));
+    let solutionHtml = this._getSolutionHtml(state);
+
+    // If solutionHtml is not null (which means that the solution in the state
+    // does exist), then add solutionHtml to _allHtmlInTheState.
+    if (solutionHtml !== null) {
+      _allHtmlInTheState.push(solutionHtml);
```

Comment:
Is it valid if the solution does not exist?

Reply:
Yes. Based on a description of the solution variable in the InteractionBackendDict interface in InteractionObjectFactory.ts, "A null 'solution' indicates that this Interaction does not have a hint or there is a hint, but no solution. A new interaction is initialised with null 'solution' and stays null until the first hint with solution is added". So, it's valid if it's null.